applications/tari_dan_app_utilities/src/base_layer_scanner.rs

```diff
@@ -264,7 +264,22 @@ impl<TAddr: NodeAddressable + 'static> BaseLayerScanner<TAddr> {
             },
             Some(end_height) => end_height,
         };
-
+        let mut scan = tip.tip_hash;
+        loop {
+            let header = self.base_node_client.get_header_by_hash(scan).await?;
+            if let Some(last_tip) = self.last_scanned_tip {
+                if last_tip == scan {
+                    // This was processed on the previous call to this function.
+                    break;
+                }
+            }
+            if header.height == end_height {
+                // This will be processed down below.
+                break;
+            }
+            self.epoch_manager.update_epoch(header.height, scan, false).await?;
```

Comment:
Rather make two calls than pass in some opaque bool. 

First call (here) adds the block. Second call (below) updates the epoch

Reply:
I did separate the calls, and also the message that is send around in the function. That way if we decide to delete this code it will be easier to do so.